tests/func-tests/perses_dynamic_crd_test.go

```diff
@@ -0,0 +1,122 @@
+package tests_test
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	persesv1alpha1 "github.com/rhobs/perses-operator/api/v1alpha1"
+
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+	tests "github.com/kubevirt/hyperconverged-cluster-operator/tests/func-tests"
+)
+
+var _ = Describe("Perses dynamic CRD gating", Label(tests.OpenshiftLabel, "perses"), func() {
+	var cli client.Client
+
+	BeforeEach(func() {
+		cli = tests.GetControllerRuntimeClient()
+	})
+
+	It("does not create resources before CRDs; creates after installing dashboards/datasources CRDs", func(ctx context.Context) {
+		// If dashboards/datasources CRDs already exist, skip to avoid interfering with a live setup.
+		if err := cli.Get(ctx, client.ObjectKey{Name: hcoutil.PersesDashboardsCRDName}, &apiextensionsv1.CustomResourceDefinition{}); err == nil {
```

Comment:
WDYT about checking that the datasource and the dashboard were deployed, if perses is installed?